frontend/src/pages/Project/Details/index.tsx

```diff
@@ -1,14 +1,131 @@
-import React from 'react';
-import { Outlet, useParams } from 'react-router-dom';
+import React, { useMemo } from 'react';
+import { Outlet, useNavigate, useParams } from 'react-router-dom';
+import { useTranslation } from 'react-i18next';
 
-import { ContentLayout, DetailsHeader } from 'components';
+import { Button, ContentLayout, DetailsHeader } from 'components';
+
+import { useAppSelector, useNotifications } from 'hooks';
+import { selectUserData } from 'App/slice';
+import { ROUTES } from 'routes';
+import { useGetProjectQuery, useAddProjectMemberMutation, useRemoveProjectMemberMutation } from 'services/project';
+import { getProjectRoleByUserName } from '../utils';
 
 export const ProjectDetails: React.FC = () => {
+    const { t } = useTranslation();
     const params = useParams();
+    const navigate = useNavigate();
     const paramProjectName = params.projectName ?? '';
+    const [pushNotification] = useNotifications();
+    const userData = useAppSelector(selectUserData);
+    
+    const { data: project } = useGetProjectQuery({ name: paramProjectName });
+    const [addMember, { isLoading: isAdding }] = useAddProjectMemberMutation();
+    const [removeMember, { isLoading: isRemoving }] = useRemoveProjectMemberMutation();
+
+    const currentUserRole = useMemo(() => {
+        if (!userData?.username || !project) return null;
+        return getProjectRoleByUserName(project, userData.username);
+    }, [project, userData?.username]);
+
+    const isProjectOwner = useMemo(() => {
```

Comment:
```suggestion
    const isProjectOwner = userData?.username === project?.owner.username;
```
Don't need to use memoization here